### ISSUE TITLE
Small Caps font feature does not work if Language is selected

### ISSUE THREAD

**Issue by borisveytsman** [latex3/fontspec#517]
## Description
Version v2.9d (the latest dev version) does not understand small caps if Language is selected

## Check/indicate
- [x] Relevant for XeTeX
- [x] Relevant for LuaTeX
- [x] Issue tracker has been searched for similar issues?

## Minimal example demonstrating the issue
```
\listfiles
\documentclass{article}
\usepackage[english]{babel}
\RequirePackage{fontspec}
\setmainfont
[
  Ligatures=TeX,
  Numbers=Lowercase,
  Extension=.otf,
  UprightFont=*-Regular,
  ItalicFont=*-Italic,
  BoldFont=*-Bold,
  BoldItalicFont=*-BoldItalic,
  Language=English,
  Script=Latin
]{EBGaramond}
\begin{document}
\textsc{aaaa23}
\end{document}
```

## Further details
The example above gives warnings
```
Package fontspec Warning: 
(fontspec)                Language 'English' not explicitly supported within
(fontspec)                font 'EBGaramond-Regular' with script 'Latin'. Check
(fontspec)                the typeset output, and if it is okay then ignore
(fontspec)                this warning. Otherwise a different font should be
(fontspec)                chosen.
```
and 
```
LaTeX Font Warning: Font shape `TU/EBGaramond(0)/m/sc' undefined
(Font)              using `TU/EBGaramond(0)/m/n' instead on input line 18.
```

Here are the package versions used
```
 *File List*
 article.cls    2023/05/17 v1.4n Standard LaTeX document class
  size10.clo    2023/05/17 v1.4n Standard LaTeX file (size option)
   babel.sty    2024/04/20 v24.4 The Babel package
 english.ldf    2017/06/06 v3.3r English support from the babel system
babel-english.tex
fontspec.sty    2024/05/07 v2.9d Font selection for XeLaTeX and LuaLaTeX
  xparse.sty    2024-03-14 L3 Experimental document command parser
   expl3.sty    2024-04-11 L3 programming layer (loader) 
l3backend-luatex.def    2024-04-11 L3 backend support: PDF output (LuaTeX)
fontspec-luatex.sty    2024/05/07 v2.9d Font selection for XeLaTeX and LuaLaTeX

 fontenc.sty    2021/04/29 v2.0v Standard LaTeX package
fontspec.cfg
  ts1cmr.fd    2023/04/13 v2.5m Standard LaTeX font definitions
 ***********
```

If I comment out Language, the bug disappears
```
\listfiles
\documentclass{article}
\usepackage[english]{babel}
\RequirePackage{fontspec}
\setmainfont
[
  Ligatures=TeX,
  Numbers=Lowercase,
  Extension=.otf,
  UprightFont=*-Regular,
  ItalicFont=*-Italic,
  BoldFont=*-Bold,
  BoldItalicFont=*-BoldItalic,
%  Language=English,
  Script=Latin
]{EBGaramond}
\begin{document}
\textsc{aaaa23}
\end{document}
```

**Comment by wspr** [comment]
Ah. Sorry about that. My test suite coverage needs improvement!

I think fontspec should check if small caps exists within any script+language combo in a preset but customisable list of defaults (e.g. Latin+Default, Default+Default,...).

If it exists there then we can still apply it and the engine will likely figure out what to do.

I'm starting to wonder if my interpretation of @khaledhosny 's advice to stop checking Script + Language was correct... maybe it would have been better to add a new set of options like `ForceLanguage` and `ForceScript` which had the new (non-checking) behaviour, and the old ones behaved as they always did.

**Comment by borisveytsman** [comment]
Since many people use babel and other interfaces, you probably want to set the old behavior as the one invoked by `\babelfont`, unless one explicitly selects the option, say,  `StrictChecking`

**Comment by khaledhosny** [comment]
This seem to be another case of #515. Diffing the long when language is specified and when it is not, I see:
```diff
-(fontspec)             <->"[EBGaramond-Regular.otf]/OT:script=latn;language=ENG
-;mapping=tex-text;"
-(fontspec)             - 'small caps'  (m/sc) with NFSS spec.:
+(fontspec)             <->"[EBGaramond-Regular.otf]/OT:script=latn;language=dfl
+t;+onum;mapping=tex-text;"
+(fontspec)             - 'small caps'  (m/sc) with NFSS spec.:
+(fontspec)             <->"[EBGaramond-Regular.otf]/OT:script=latn;language=dfl
+t;+onum;+smcp;mapping=tex-text;"
```

So fontspec is not requesting the feature when the language is set.

**Comment by khaledhosny** [comment]
I have fontspec 2.9c, though, which I think does not have the fix for #515.

**Comment by khaledhosny** [comment]
If I understand correct, fontspec wants to know if the font supports smcp feature to set up NFSS smallcap font. That is indeed different from the user explicitly asking for a feature. I think in this case checking for smallcaps under any script/language combination is probably the safest approach.

You can keep the existing behavior if both specified script and language exist, and check any combination if the specified ones do not exist.
